src/tiledb/cloud/sql/db_api_exceptions.py

```diff
@@ -0,0 +1,38 @@
+class Error(Exception):
+    """Base class for exceptions in the DB API 2.0 implementation."""
+
+
+class InterfaceError(Error):
+    """Raised when the database encountered a programming error."""
+
+
+class DatabaseError(Error):
+    """Raised when the database encountered a programming error."""
+
+
+class _Warning(Exception):
+    """Exception for important warnings."""
```

Comment:
While this will shadow the built-in `Warning` type, the DB API specifies that it should be named that way, so just `class Warning` here.